recognition/app/presentation/securityrule/IntrusionRuleSerializer.py

```diff
@@ -0,0 +1,57 @@
+from datetime import datetime
+from typing import List
+
+from app.domain.anomaly.core import ObjectClass
+from app.domain.device.core import DeviceId, DeviceType
+from app.domain.monitoring.core import ContactType, Contact
+from app.domain.securityrule.core import IntrusionRule
+
+
+class IntrusionRuleSerializer:
+
+    @staticmethod
+    def deserialize(data: dict) -> IntrusionRule:
+        security_rule_id = data["_id"]
+        device_type: DeviceType = DeviceType(data["deviceId"]["type"])
+        device_id: DeviceId = DeviceId(device_type, data["deviceId"]["code"])
+        creator_id: str = data["creatorId"]
+        description: str = data["description"]
+        object_class: ObjectClass = ObjectClass(data["objectClass"])
+        contacts_to_notify: List[Contact] = [
+            Contact(item["value"], ContactType(item["type"]))
+            for item in data["contactsToNotify"]
+        ]
+
+        from_date: datetime = datetime.fromisoformat(
+            data["from"].replace("Z", "+00:00")
+        )
+        to_date: datetime = datetime.fromisoformat(data["to"].replace("Z", "+00:00"))
+        return IntrusionRule(
+            object_class,
+            security_rule_id,
+            device_id,
+            creator_id,
+            contacts_to_notify,
+            description,
+            from_date,
+            to_date,
+        )
+
+    @staticmethod
+    def serialize(security_rule: IntrusionRule) -> dict:
+        return {
+            "_id": security_rule.security_rule_id,
+            "deviceId": {
+                "type": security_rule.device_id.type.value,
+                "code": security_rule.device_id.code,
+            },
+            "creatorId": security_rule.creator_id,
+            "description": security_rule.description,
+            "objectClass": security_rule.object_class.value,
+            "contactsToNotify": [
```

Comment:
we need to check how is the contacs field in frontend and in other services, i'll add this to my todo to uniform them to 'contacts' or 'contactsToNotify', tell me what you prefer

Reply:
contactsToNotify is fine